base/FieldUtilities.F90

```diff
@@ -0,0 +1,53 @@
+#include "MAPL_Generic.h"
+
+module MAPL_FieldUtilities
+use ESMF
+use MAPL_BaseMod, only: MAPL_Undef
+use MAPL_ErrorHandlingMod
+
+implicit none
+private
+
+public check_if_field_undef
+
+contains
+
+function check_if_field_undef(field,rc) result(field_is_undef)
+   logical :: field_is_undef
```

Comment:
This name is not well suited to a function as opposed to a subroutine.   I prefer something like:
```suggestion
function is_field_undef(field,rc) result(field_is_undef)
   logical :: is_field_undef
```